### PR TITLE
ZEPPELIN-1037 Enable Kerberos support in Livy Interpreter

### PR DESCRIPTION
**Pull request by r-kamath**
### What is this PR for?
This PR is for enabling kerberos support in Livy interpreter. Also introduced two new Livy interpreter properties called keytab and principal to configure. 

### What type of PR is it?
Improvement

### What is the Jira issue?
ZEPPELIN-1037

### How should this be tested?
Kerberize the Livy server and configure the Livy interpreter with appropriate keytab and principal.

### Questions:
* Does the licenses files need update? no
* Is there breaking changes for older versions? no
* Does this needs documentation? yes

### CODE REVIEW

File: livy/src/main/java/org/apache/zeppelin/livy/LivyHelper.java [hunk 1]
```diff
@@ -69,15 +68,16 @@ public Integer createSession(InterpreterContext context, String kind) throws Exc
...
 
       String json = executeHTTP(property.getProperty("zeppelin.livy.url") + "/sessions",
-          "POST",
-          "{" +
-              "\"kind\": \"" + kind + "\", " +
-              "\"conf\": " + confData + ", " +
-              "\"proxyUser\": \"" + context.getAuthenticationInfo().getUser() +
-              "\"}",
-          context.getParagraphId()
+              "POST",
```
Comment (jongyoul) [inline livy/src/main/java/org/apache/zeppelin/livy/LivyHelper.java]: First of all, it seems to fix your indentation.

**Comment by jongyoul** [comment]
@r-kamath There's are two kind of http-client libraries. Can you replace one to another? I don't know if it's good that we use Spring framework or not, but it looks like a problem that we have two different libraries with same function. How about it?

File: livy/src/main/java/org/apache/zeppelin/livy/LivyHelper.java [hunk 2]
```diff
@@ -69,15 +68,16 @@ public Integer createSession(InterpreterContext context, String kind) throws Exc
...
+              "{" +
+                "\"kind\": \"" + kind + "\", " +
+                "\"conf\": " + confData + ", " +
+                "\"proxyUser\": " + (user == null ? null : "\"" + user + "\"") +
```
Comment (felixcheung) [inline livy/src/main/java/org/apache/zeppelin/livy/LivyHelper.java]: should we check if user is an empty string here?

Comment (r-kamath) [inline livy/src/main/java/org/apache/zeppelin/livy/LivyHelper.java]: fixed

**Comment by r-kamath** [comment]
@jongyoul fixed the indentation and removed apache http client dep.

**Comment by r-kamath** [comment]
CI failed at one of the build matrix

**Comment by jongyoul** [comment]
@r-kamath Thanks for fixing it.

**Comment by prabhjyotsingh** [comment]
Have tested it, it works on a Kerberos enabled system.

**Comment by prabhjyotsingh** [comment]
Merging this if no more discussion.

**Pull request closed without merging**